File: src/com/crossbowffs/quotelock/modules/natune/NatuneQuoteModule.java
package com.crossbowffs.quotelock.modules.natune;

import android.content.ComponentName;
import android.content.Context;
import com.crossbowffs.quotelock.R;
import com.crossbowffs.quotelock.api.QuoteData;
import com.crossbowffs.quotelock.api.QuoteModule;
import com.crossbowffs.quotelock.utils.IOUtils;
import org.jsoup.Jsoup;
import org.jsoup.nodes.Document;
import org.jsoup.nodes.Element;

public class NatuneQuoteModule implements QuoteModule {
    @Override
    public String getDisplayName(Context context) {
        return context.getString(R.string.module_natune_name);
    }

    @Override
    public ComponentName getConfigActivity(Context context) {
        return null;
    }

    @Override
    public int getMinimumRefreshInterval(Context context) {
        return 0;
    }

    @Override
    public boolean requiresInternetConnectivity(Context context) {
        return true;
    }

    @Override
    public QuoteData getQuote(Context context) throws Exception {
        String html = IOUtils.downloadString("https://natune.net/zitate/Zufalls5");
        Document document = Jsoup.parse(html);
        Element quoteLi = document.select(".quotes > li").first();
        String quoteText = quoteLi.getElementsByClass("quote_text").first().text();
        String quoteAuthor = quoteLi.getElementsByClass("quote_author").first().text();
        return new QuoteData(quoteText, "― " + quoteAuthor);
    }
}

File: app/src/main/java/com/crossbowffs/quotelock/data/modules/natune/NatuneQuoteModule.kt
package com.crossbowffs.quotelock.data.modules.natune

import android.content.Context
import com.crossbowffs.quotelock.data.api.QuoteData
import com.crossbowffs.quotelock.data.api.QuoteModule
import com.crossbowffs.quotelock.data.api.QuoteModule.Companion.CHARACTER_TYPE_LATIN
import com.crossbowffs.quotelock.utils.fetchXml
import com.yubyf.quotelockx.R

class NatuneQuoteModule : QuoteModule {
    override fun getDisplayName(context: Context): String {
        return context.getString(R.string.module_natune_name)
    }

    override fun getConfigRoute(): String? = null

    override fun getMinimumRefreshInterval(context: Context): Int {
        return 0
    }

    override fun requiresInternetConnectivity(context: Context): Boolean {
        return true
    }

    @Throws(Exception::class)
    override suspend fun Context.getQuote(): QuoteData {
        val document = httpClient().fetchXml("https://natune.net/zitate/Zufalls5")
        val quoteLi = document.select(".quotes > li").first()
        val quoteText = quoteLi?.getElementsByClass("quote_text")?.first()?.text().orEmpty()
        val quoteAuthor = quoteLi?.getElementsByClass("quote_author")?.first()?.text().orEmpty()
        return QuoteData(
            quoteText = quoteText,
            quoteSource = "",
            quoteAuthor = quoteAuthor,
            provider = "natune"
        )
    }

    override val characterType: Int
        get() = CHARACTER_TYPE_LATIN
}